graph-commons/src/main/scala/io/renku/http/client/AccessToken.scala

```diff
@@ -28,40 +28,60 @@ import java.util.Base64
 
 sealed trait AccessToken extends Any with StringTinyType with Sensitive
 
+sealed trait UserAccessToken extends Any with AccessToken
+
 object AccessToken {
 
-  final class PersonalAccessToken private (val value: String) extends AnyVal with AccessToken
+  final class PersonalAccessToken private (val value: String) extends AnyVal with UserAccessToken
   object PersonalAccessToken
       extends TinyTypeFactory[PersonalAccessToken](new PersonalAccessToken(_))
       with NonBlank[PersonalAccessToken]
 
-  final class OAuthAccessToken private (val value: String) extends AnyVal with AccessToken
+  final class OAuthAccessToken private (val value: String) extends AnyVal with UserAccessToken
   object OAuthAccessToken
       extends TinyTypeFactory[OAuthAccessToken](new OAuthAccessToken(_))
       with NonBlank[OAuthAccessToken]
 
+  final class ProjectAccessToken private (val value: String) extends AnyVal with AccessToken
+  object ProjectAccessToken
+      extends TinyTypeFactory[ProjectAccessToken](new ProjectAccessToken(_))
+      with NonBlank[ProjectAccessToken]
+
   private val base64Decoder = Base64.getDecoder
   private val base64Encoder = Base64.getEncoder
 
   implicit val accessTokenEncoder: Encoder[AccessToken] = {
+    case ProjectAccessToken(token) =>
+      Json.obj("projectAccessToken" -> Json.fromString(new String(base64Encoder.encode(token.getBytes(UTF_8)), UTF_8)))
     case OAuthAccessToken(token) =>
```

Comment:
Now the name `OAuth…` seems a bit too generic. It is only for a user token used. If understood correctly the "project token" is an oauth token as well (being used in an oauth flow)? (nit picking obviously :))

Reply:
True. Perhaps, it'd be better to name it `UserOAuthAccessToken`? What do you think?

Reply:
Yes, that sounds good 👍🏼